python/whylogs/api/logger/logger.py

```diff
@@ -98,13 +102,22 @@ def log(
 
         # If segments are defined use segment_processing to return a SegmentedResultSet
         if active_schema and active_schema.segments:
-            return segment_processing(active_schema, obj, pandas, row, self._segment_cache)
+            segmented_results = segment_processing(active_schema, obj, pandas, row, self._segment_cache)
+            _extract_common_profile_metadata(
+                segmented_results.metadata, trace_id=trace_id, tags=tags, segment_key_values=segment_key_values
+            )
+            return segmented_results
 
         profiles = self._get_matching_profiles(obj, pandas=pandas, row=row, schema=active_schema)
 
         for prof in profiles:
             prof.track(obj, pandas=pandas, row=row, execute_udfs=False)
-            prof._metadata
```

Comment:
Was old line 107 a bug?

Reply:
yes! A casualty of splitting previous PRs from larger changes with the _metadata.